src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/GeneratedComInterfaceAttributeAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#nullable enable
+namespace Microsoft.Interop
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class GeneratedComInterfaceAttributeAnalyzer : DiagnosticAnalyzer
+    {
+        private const string _generatedComInterfaceAttributeName = "GeneratedComInterfaceAttribute";
+        private const string _generatedComInterfaceAttributeNamespace = "System.Runtime.InteropServices.Marshalling";
+
+        private const string _interfaceTypeAttributeName = "InterfaceTypeAttribute";
+        private const string _interfaceTypeAttributeNamespace = "System.Runtime.InteropServices";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+            = ImmutableArray.Create(GeneratorDiagnostics.InterfaceTypeNotSupported);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.EnableConcurrentExecution();
+            context.RegisterSymbolAction((context) =>
+            {
+                INamedTypeSymbol typeSymbol = (INamedTypeSymbol)context.Symbol;
+                if (typeSymbol.TypeKind != TypeKind.Interface)
+                    return;
+                ImmutableArray<AttributeData> customAttributes = typeSymbol.GetAttributes();
+                if (customAttributes.Length == 0)
+                    return;
+
+                // Interfaces should not have both GeneratedComInterfaceTypeAttribute and InterfaceTypeAttribute
+                if (GetGeneratedComInterfaceAttribute(typeSymbol, out AttributeData? generatedComInterfaceAttribute)
```

Comment:
```suggestion
                if (GetGeneratedComInterfaceAttribute(typeSymbol, out AttributeData? _)
```
We don't seem to actually be using it (at least right now).

Reply:
can also just do `out _` fwiw :)